### ISSUE TITLE
Make it easier to migrate from Jupyter Notebook to Metaflow Steps

### ISSUE THREAD

**Issue by jimmycfa** [Netflix/metaflow#309]
Most data scientist do their development in Jupyter Notebooks. Recommendation would be to use something like [Kale](https://github.com/kubeflow-kale/kale) to be able to tag cells in Jupyter Notebook and for them to be automatically turned into Metaflow steps to make the Notebook to Metaflow Steps process easier.

This idea was really taken from [this](https://medium.com/kubeflow/automating-jupyter-notebook-deployments-to-kubeflow-pipelines-with-kale-a4ede38bea1f) article.

**Comment by tuulos** [comment]
Thanks for the links @jimmycfa. We have been thinking about the topic of a long time. While technically grabbing code from cells is quite doable, it is less clear how one should version control the code, define a complex DAG, `resume` executions etc.

Metaflow is definitely meant to be used in conjunction with notebooks. At Netflix, we use notebooks to inspect results using the Client API. We have also a @notebook decorator internally, which we are planning to open-source, which automatically executes notebooks when a Metaflow step finishes.

Long story short, this idea is on our radar. Stay tuned!

**Comment by lsimoneau** [comment]
Hi! Has there been any further thinking on this? We're in a similar position to @jimmycfa and it would be really useful to have some way of more easily moving code from notebooks into Metaflow.

**Comment by jixianyihao** [comment]
We currently have hundreds of tasks on our platform that require bulk updates. Managing them solely through commands makes it difficult to integrate with other systems. Is it possible to address this issue in the next version? Introducing a class and directly invoking a specific method within the class to initiate a Flow would be a significant enhancement.

**Comment by tuulos** [comment]
@jixianyihao good feedback. There are various ways to do such updates today. Can you share more details about what you need to update: Parameters, decorators, business logic, or something else?

**Comment by tuulos** [comment]
There's a nice [notebook integration available now with `NBRunner`](https://docs.metaflow.org/metaflow/managing-flows/notebook-runs)
